subsys/llext/llext.c

```diff
@@ -709,48 +909,67 @@ int llext_load(struct llext_loader *ldr, const char *name, struct llext **ext)
 		}
 
 		ldr->hdr = ehdr;
-		ret = do_llext_load(ldr, *ext);
-		break;
-	default:
-		LOG_ERR("Unsupported elf file type %x", ehdr.e_type);
-		*ext = NULL;
-		return -EINVAL;
-	}
+		ret = do_llext_load(ldr, *ext, do_local);
+		if (ret < 0) {
+			k_heap_free(&llext_heap, *ext);
+			*ext = NULL;
+			return ret;
+		}
 
-	if (ret == 0) {
 		strncpy((*ext)->name, name, sizeof((*ext)->name));
 		(*ext)->name[sizeof((*ext)->name) - 1] = '\0';
+		(*ext)->use_count++;
+
 		sys_slist_append(&_llext_list, &(*ext)->_llext_list);
```

Comment:
Since we have a mutex protecting the removal we can reuse it presumably to protect appending a module here

Reply:
the purpose of locks has to be strictly defined. So far `llext_lock` is used to protect reference counting. If we also want to use it to protect the list of modules, it has to be carefully considered and - if accepted - properly documented. I haven't done that yet, therefore it's still a "TODO" / "FIXME." In fact so far we don't use that list of modules, but we probably will need it when modules begin using symbols, exported by other modules.